### PR TITLE
feature: code blocks w/ plugin

### PR DESCRIPTION
**Pull request by howebenhowe**
We are going with this approach!

_Test Plan_
create code block in a pub and play with them generally. Steps below. NB: the code blocks behave as a text block, which is to say, they are not deletable if they are the only text block in the document just as the only `paragraph` block would not be deletable. We can change this behavior, but it would be a departure from block behavior to accommodate image-block-like behavior.
Make sure to hit each of the below:
+ create a code block by typing three backticks to enter it
+ create a code block by clicking the lego-block-shaped codeblock toggle button in the formatting bar
+ create a code block by selecting `code` from the dropdown at the left edge of the formatting bar
+ exit the codeblock by pressing `return` from an empty line in the codeblock
+ exit the codeblock by navigating to the right from the last character (per NB above, this is impossible if there isn't a following text block to navigate into)
+ while inside a codeblock, select different language syntax highlighting grammars, then reload the page
+ copy + paste into/out of codeblocks
+ delete codeblocks with backspace (provided there is a text node above the codeblock in the document)

TODO:
+ implement the language selector dropdown component using Blueprint components for visual consistency, using the `createSelect` api from the vendor module.
+ PDF export

### CODE REVIEW

**Comment by 3mcd** [comment]
Looks and works great! 🎉  Tested a few other scenarios like collaboration and undo/redo and they worked great. Waiting for styling tweaks to approve.

File: client/components/Editor/plugins/code.ts
Comment (idreyn) [inline client/components/Editor/plugins/code.ts:25]: `bp3-icon-double-caret-vertical` is having no effect as far as I can see (Firefox). I also think that we should use `caret-down` for consistency with other dropdowns in Blueprint.

File: client/components/Editor/plugins/code.ts
Comment (idreyn) [inline client/components/Editor/plugins/code.ts:56]: Is this a note to self about this code, or is it about the code block itself?

**Review by idreyn — APPROVED** [review]
LGTM, I have one usability concern I'd like to see addressed before landing — we need to make sure the `<PubInlineMenu>` isn't showing up while you're inside a code block. We'll have to add another clause to this chunk of code here:

https://github.com/pubpub/pubpub/blob/8064b262695e9bc959589c8e222a5936ca3664ca/client/containers/Pub/PubDocument/PubInlineMenu.tsx#L30

I'd recommend breaking your new logic along with the existing logic into a helper function and then computing it inside a `useMemo()` for best performance.

image

My other thought is, maybe the font size for code blocks should be a few px smaller?

**Review by 3mcd — APPROVED** [review]

**Pull request merged**
